app/src/main/java/com/andrew/tasky/presentation/event_detail/EventDetailFragment.kt

```diff
@@ -0,0 +1,723 @@
+        }
+    }
+
+    private fun setupTitleLayout() {
+        binding.apply {
+
+            collectLatestLifecycleFlow(viewModel.title) { title ->
+                titleTextView.text = title
+            }
+
+            if (!isAttendee) {
+                collectLatestLifecycleFlow(viewModel.isInEditMode) { isEditing ->
+                    editTitleButton.isVisible = isEditing
+                    editTitleButton.isEnabled = isEditing
+                    titleTextView.isEnabled = isEditing
+                }
+                titleTextView.setOnClickListener {
+                    navigateToEditFragment(EditType.TITLE)
+                }
+                editTitleButton.setOnClickListener {
+                    navigateToEditFragment(EditType.TITLE)
+                }
+            } else {
+                editTitleButton.isVisible = false
+                editTitleButton.isEnabled = false
+                titleTextView.isEnabled = false
+            }
+        }
+    }
+
+    private fun setupDescriptionLayout(){
+        binding.apply {
+
+            collectLatestLifecycleFlow(viewModel.description) { description ->
+                descriptionTextView.text = description
+            }
+            
+            if (!isAttendee){
+                collectLatestLifecycleFlow(viewModel.isInEditMode) { isEditing ->
+                    editDescriptionButton.isVisible = isEditing
+                    editDescriptionButton.isEnabled = isEditing
+                    descriptionTextView.isEnabled = isEditing
+                }
+                editDescriptionButton.setOnClickListener {
+                    navigateToEditFragment(EditType.DESCRIPTION)
+                }
+                descriptionTextView.setOnClickListener {
+                    navigateToEditFragment(EditType.DESCRIPTION)
+                }
+            }
+            else{
+                editDescriptionButton.isVisible = false
+                editDescriptionButton.isEnabled = false
+                descriptionTextView.isEnabled = false
+            }
+        }
+    }
+
+    private fun setupPhotoLayout(){
+        fun addPhoto(){
+            addPhotoSearchForResult.launch("image/*")
+        }
+
+        fun openPhoto(index: Int) {
+            setFragmentResultListener("REQUEST_KEY"){
+                resultKey, bundle -> if(resultKey == "REQUEST_KEY"){
+                    val photoToDelete = bundle.getInt("DELETE_PHOTO_INDEX")
+                    viewModel.deletePhoto(photoToDelete)
+                }
+            }
+            navController.navigate(
+                EventDetailFragmentDirections
+                    .actionEventDetailFragmentToPhotoDetailFragment(
+                        viewModel.photos.value[index].toString(), index
+                    )
+            )
+        }
+
+        binding.apply {
+            collectLatestLifecycleFlow(viewModel.photos) { photoList ->
+                addPhotoLayout.isVisible = !(isAttendee && photoList.isEmpty())
```

Comment:
If you negate a whole boolean condition, it's equivalent to negating all single parts and swapping && to || or vice versa:
`!(isAttendee && photoList.isEmpty()) == !isAttendee || photoList.isNotEmpty()`

The latter is a bit easier to process in your brain IMO